A78-grpc-metrics-wrr-pf-xds.md

```diff
@@ -0,0 +1,148 @@
+
+The following metrics will be exported:
+
+| Name          | Type  | Unit  | Labels  | Description |
+| ------------- | ----- | ----- | ------- | ----------- |
+| grpc.pick_first.disconnections | Counter | {disconnections} | grpc.target | Number of times the selected subchannel becomes disconnected. |
+| grpc.pick_first.connections_succeeded | Counter | {connections} | grpc.target | Number of times a subchannel is successfully connected. |
+| grpc.pick_first.connection_attempts_failed | Counter | {attempts} | grpc.target | Number of failed connection attempts. |
+
+### XdsClient
+
+The XdsClient component was originally described in [A27].  Note that in
+[A71], we are moving from a single global XdsClient instance to a
+separate global XdsClient instance for each channel target.  The
+proposed metric schema here reflects that change.
+
+The following labels may be used for XdsClient metrics:
+
+| Name        | Description |
+| ----------- | ----------- |
+| grpc.target | For clients, indicates the target of the gRPC channel in which WRR is used (i.e., the same as the attribute defined in [A66]). For servers, will be the string "#server". |
+| grpc.xds_server | The name of the xDS server with which the XdsClient is communicating. |
+| grpc.xds_authority | The xDS authority.  The value will be "old" for old-style non-xdstp resource names. |
+| grpc.xds_cache_state | Indicates the cache state of an xDS resource.  The value will be one of: <ul><li>"requested": The resource has been requested from the xDS server but has not yet been received.<li>"does_not_exist": The server has indicated that the resource does not exist.<li>"acked": The resource has been received and is valid.<li>"nacked": The resource was received but was not valid.<li>"nacked_but_cached": There is a version of the resource cached, but the most recent update of the resource was invalid.</ul> |
+| grpc.xds_resource_type | Indicates an xDS resource type, such as "envoy.config.listener.v3.Listener". |
+
+The following metrics will be exported:
+
+| Name          | Type  | Unit  | Labels  | Description |
+| ------------- | ----- | ----- | ------- | ----------- |
+| grpc.xds_client.xds_server.connected | Gauge | {bool} | grpc.target, grpc.xds_server | Whether or not the xDS client currently has a working ADS stream to the xDS server.  For a given server, this will be set to 0 when we have a connectivity failure or when the ADS stream fails without seeing a response message, as per [A57].  It will be set to 1 when we receive the first response on an ADS stream. |
+| grpc.xds_client.xds_server.updates | Counter | {updates} | grpc.target, grpc.xds_server, grpc.xds_resource_type | A counter of resource updates from the xDS server.  Note that this is a count of resources, not response messages; if a response message contains two resources, then we will increment the counter twice. |
```

Comment:
Those two do not have the grpc.xds_authority, is that an omission? I don't see why it would not apply.

Reply:
OK, that makes sense.